textproto/header_test.go

```diff
@@ -205,6 +205,23 @@ func TestReadHeader(t *testing.T) {
 	}
 }
 
+const testInvalidHeader = "Received: from example.com by example.org\r\n" +
+	"Received: from localhost by example.com\r\n" +
+	"To: Taki Tachibana <taki.tachibana@example.org>\r\n" +
+	"From: Mitsuha Miyamizu <mitsuha.miyamizu@example.com>\r\n" +
+	"not valid: example\r\n"
+
+func TestInvalidHeader(t *testing.T) {
+	r := bufio.NewReader(strings.NewReader(testInvalidHeader))
+	_, err := ReadHeader(r)
+	if err == nil {
+		t.Errorf("No error thrown")
+		// Header is invalid because header name contains a space
+	} else if !strings.HasSuffix(err.Error(), "not valid: example\r\n") {
```

Comment:
I'd prefer not to check for the error message. The error message is a human-readable string that shouldn't be recognized by machines and that is not guaranteed to be stable.

Reply:
I thought about that too — what would be the best way to make sure that it failed on this specific line and not before? Maybe have the test case only consist of a malformed header?